controllers/v1/solutions.js

```diff
@@ -0,0 +1,113 @@
+const config = require('../../config/config');
+const rp = require('request-promise');
+const request = require('request');
+const model = require('../../db');
+const helperFunc = require('../../helper/chart_data');
+const authService = require('../../middleware/authentication_service');
+
+/**
+   * @api {post} /dhiti/api/v1/solutions/list 
+   * Solutions List 
+   * @apiVersion 1.0.0
+   * @apiGroup Solutions
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiParamExample {json} Request-Body:
+* {
+  "entityId": "",
+  "entityType": "",
+  "programId": ""
+* }
+   * @apiSuccessExample {json} Success-Response:
+*     HTTP/1.1 200 OK
+*     {
+       "result": true,
+       "data": [
+           {
+               "solutionId": "",
+               "solutionName": ""
+           }
+       ]
+*     }
+   * @apiUse errorBody
+   */
+
+
+//Controller for listing solution Names
+exports.list = async function (req, res) {
+
+    if (!req.body.entityId || !req.body.entityType || !req.body.programId || !req.body.type) {
+        res.status(400);
+        var response = {
+            result: false,
+            message: 'entityId,entityType,programId and type are required fields'
+        }
+        res.send(response);
+    }
+    else {
+
+        let query;
+
+        if (req.body.reportType == "my") {
```

Comment:
@deepakr123 - Please evaluate of both responses can be combined into 1 response.

Reply:
resolved